openproblems/tasks/spatial_decomposition/datasets/_utils.py

```diff
@@ -0,0 +1,101 @@
+from .._utils import obs_means
+from anndata import AnnData
+from scipy.sparse import csr_matrix
+
+import numpy as np
+import scanpy as sc
+
+
+# pass the reference data
+def generate_synthetic_dataset(adata: AnnData, sim_type: str = "avg"):
```

Comment:
Can you please write a more detailed docstring here? Here's a template:
```
    """Short summary.

    Parameters
    ----------
    adata : AnnData
        Description of parameter `adata`.
    sim_type : str
        Description of parameter `sim_type`.

    Returns
    -------
    type
        Description of returned object.

    """
```

Reply:
Also the whole function could really use some more detailed commenting

Reply:
added a bit more, but again I think we should heavily rewrite this (but would need inputs from others)